lib/internal/repl.js

```diff
@@ -121,22 +121,43 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
     if (data) {
       repl.history = data.split(/[\n\r]+/, repl.historySize);
     } else if (oldHistoryPath) {
-      // Grab data from the older pre-v3.0 JSON NODE_REPL_HISTORY_FILE format.
-      repl._writeToOutput(
+      if (oldHistoryPath === historyPath) {
```

Comment:
Correct me if I'm wrong, but I think it could be shortened to:

```js
if (data) {
} else if (oldHistoryPath === historyPath) {
} else if (oldHistoryPath) {
}
```